File: cielo24_PHP/tests/ActionsTest.php
<?php

require_once("../src/cielo24/Actions.php");
require_once("Config.php");

class ActionsTest extends PHPUnit_Framework_TestCase
{
    protected $actions;
    protected $config;
    protected $apiToken = null;
    protected $secureKey = null;

    public function setUp()
    {
        $this->config = new Config();
        $this->actions = new Actions($this->config->serverUrl);
        if ($this->apiToken == null) {
            $this->apiToken = $this->actions->login($this->config->username, $this->config->password, true);
        }
        if ($this->secureKey == null) {
            $this->secureKey = $this->actions->generateAPIKey($this->apiToken, $this->config->username, true);
        }
    }
}

File: cielo24_Python/cielo24/test/actions_test.py
# encoding: utf-8
import logging
from unittest import TestCase
from cielo24.actions import Actions
from cielo24.web_utils import WebUtils
import config as config


class ActionsTest(TestCase):

    @classmethod
    def setUpClass(cls):
        # Make WebUtils logger output to stdout
        cls.console_handler = logging.StreamHandler()
        cls.old_level = WebUtils.LOGGER.level
        WebUtils.LOGGER.level = logging.DEBUG
        WebUtils.LOGGER.addHandler(cls.console_handler)

    @classmethod
    def tearDownClass(cls):
        # Restore WebUtils logger settings
        WebUtils.LOGGER.level = cls.old_level
        WebUtils.LOGGER.removeHandler(cls.console_handler)

    def setUp(self):
        self.actions = Actions(config.server_url)
        # Start with a fresh session each time
        self.api_token = self.actions.login(config.username, config.password, None, True)
        self.secure_key = self.actions.generate_api_key(self.api_token, config.username, False)
